addons/account/models/account_move.py

```diff
@@ -5224,6 +5224,8 @@ def _message_post_after_hook(self, new_message, message_values):
         # EXTENDS mail mail.thread
         # When posting a message, check the attachment to see if it's an invoice and update with the imported data.
         res = super()._message_post_after_hook(new_message, message_values)
+        if not self.env.user._is_internal():
+            return res
```

Comment:
Note: this prevents a crash when posting a message with attachments as portal user on an invoice.
Before this PR attachments were added _after_ `message_post` with custom code in portal, so the list of attachment was always empty at this step. It is no longer empty now so we need to check differently.